lib/hmr.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const sysPath = require('path');
+const logger = require('loggy');
+
+const isEnabled = config => config.hot && config.env.indexOf('production') === -1;
+
+const checkAutoReload = (cfg, plugins) => {
+  const hasAutoReload = plugins.compilers.find(compiler => compiler.brunchPluginName === 'auto-reload-brunch');
+  if (!hasAutoReload) {
+    throw new Error("Hot Module Reloading should only be used with 'auto-reload-brunch'");
+  } else if (!hasAutoReload.supportsHMR) {
+    throw new Error("Currently used version of 'auto-reload-brunch' does not support Hot Module Reloading");
+  }
+};
```

Comment:
Or maybe just specifying a `peerDependency` for hmr on newer version of auto-reload-brunch will do